sandbox/Devis/Keyvault.tf

```
@@ -0,0 +1,32 @@
+resource "azurerm_key_vault" "kv_azure_demo_01" {
+  lifecycle {
+    prevent_destroy = true
+  }
+
+  name                = "kv-azure-demo-01"
+  resource_group_name = azurerm_resource_group.sandbox.name
+  location            = azurerm_resource_group.sandbox.location
+  tenant_id           = data.azurerm_client_config.current.tenant_id
+  sku_name            = "standard"
+  tags                = local.default_tags
+}
+
+resource "azurerm_key_vault_secret" "username" {
+  key_vault_id = azurerm_key_vault.kv_azure_demo_01.id
+  name         = "username"
+  value        = "azure_demo_user"
+}
+
+resource "random_string" "random_key_generator" {
```

if multiple resources with different types are for the same purpose, then the resource name could be kept same is all for easy reference
```suggestion
resource "random_string" "password" {
```